### ISSUE TITLE
Revamping onboarding user experience

### ISSUE THREAD

**Issue by virgile-dev** [DemocracyEarth/sovereign#238]
# This is a Feature Proposal)

#### :tophat: Description
The onboarding experience for Sovereign was lacking a proper landing page to give some context to the user registration.

Here is **what we are proposing** after the first iteration :

<img width="1825" alt="onboarding" src="https://user-images.githubusercontent.com/11473995/36154272-312ea304-10d1-11e8-833a-dd899952248f.png">

Here is **what we have now** : 
![actual](https://user-images.githubusercontent.com/11473995/36154416-a8e75ab2-10d1-11e8-9b42-a1194359b214.png)

**Vote on the design proposal** : https://vote.democracy.earth/tag/design

**Comment by aecc** [comment]
They look really nice!, maybe you can explain what are the 4 new screens? Specifically the 4th one and how to reach there. There is a small typo in "D. Be Informed" btw, word "tranparent"

**Comment by virgile-dev** [comment]
@aecc just did ! Thanks for the typo detection.

**Comment by SFSandra** [comment]
Aesthetically very pleasing, a clean and warm design. As a user I wonder which screen enables me to see votes I've cast in different elections I've participated in...related to @aecc 's point, above.

**Comment by AlexJupiter** [comment]
@SFSandra I totally agree that a way to enable someone to see the votes they've cast in different elections would be useful. What we're proposing above though is before someone has even signed up. 

After this, I'd love to move onto redesigning other key parts of the product like what you propose :).

**Comment by virgile-dev** [comment]
Hi everyone !

It seems that people are quite happy with what is being proposed :) 
![capture d ecran 2018-02-15 a 10 46 45](https://user-images.githubusercontent.com/11473995/36250287-a68f91f4-123d-11e8-8187-16a2976eec3d.png)

Let's start coding. How you guys want to proceed ? Should Piotr code the whole thing at once or should we break it into little pieces ?

@AlexJupiter what should be the next design iteration ?
Personally, For a while I've been frustrated with this https://github.com/DemocracyEarth/sovereign/issues/205

**Comment by thearkadia** [comment]
@virgile-dev what's the recommended way of getting the Mobile UI to load? I'd like to test it on our project https://github.com/thearkadia/The_Ark

**Comment by virgile-dev** [comment]
@thearkadia maybe @santisiri can answer this.

**Comment by piotrf** [comment]
Getting back to this starting next week. Can anyone confirm that there has been no change of spec and we’re still building the thing as per screenshots above? cc: @virgile-dev, @AlexJupiter?

**Comment by AlexJupiter** [comment]
@piotrf exciting! :). Yep I believe there is no change of spec and the designs above are your latest ones.

**Comment by virgile-dev** [comment]
Hey @piotrf @AlexJupiter , 
Did you guys think of how these designs will coexist with the user modals for account editing, forgot password etc... ? 
![user-modals](https://user-images.githubusercontent.com/11473995/38546867-083629fa-3cae-11e8-92b9-79412050b09c.png)
These are triggered by these : dashed-eye, user-avatar
![nav](https://user-images.githubusercontent.com/11473995/38546937-3ae200ae-3cae-11e8-902f-01dc45023e06.png)

**Comment by AlexJupiter** [comment]
@virgile-dev for the first stage we could just implement the on boarding as initially proposed, and then come back to the account editing later? What do you think @piotrf? I guess I'm keen to get this implemented, I think it will really help to explain to users what Sovereign is all about the first time they hear about it.

Also, just to confirm, we are cutting the ability to sign up anonymously for now yes?

**Comment by virgile-dev** [comment]
We might implement anonymous login this week end  with Metamask #279

**Comment by AlexJupiter** [comment]
Exciting to be working with @LucasIsasmendi to be implementing this. Here are the links to the Figma file: https://www.figma.com/file/bmy63a4jEcPfzFPmGn5vTzLN/Onboarding-v1

**Comment by piotrf** [comment]
Who’s best to walk me through the front-end a little bit? I need to know where I can change what not to break things.

In any case, I’d probably like to keep all the current components the way they are so to have them available when logged in (to answer your question @virgile-dev) but I’d like to set up an entirely new React `auth` component with its new children to display when logged out (is this even reasonable?). So:

1. I’d create `auth.jsx` with `signup.jsx`, `signin.jsx`, `recovery.jsx`, ecc. as children
2. I’d load the new `auth` if signed out (where and how should that `if` statement look like, anyone?)
3. I’d then try to hook up the new auth to the auth functions we already have

It would be helpful if, whoever has the know-how, would tell me how to set up a clean slate React component that displays `Hello world` when logged out, and I can then take it from there.

**Comment by piotrf** [comment]
Btw, created this branch to work on: https://github.com/DemocracyEarth/sovereign/tree/feature/238-onboarding

**Comment by piotrf** [comment]
Nvm, I think I figured it out. I now have the app display contents of `Onboarding.jsx` when logged out. Can carry on.

**Comment by AlexJupiter** [comment]
@lucasIsasmendi might be able to help you out if any further blockers arise. At least we spent some time on this together _trying_ to make some progress.

**Comment by AlexJupiter** [comment]
@piotrf how are you progressing with this?

**Comment by piotrf** [comment]
Assembling the views that look like we want them to look but do nothing. I’ll ping you once I have them, then I may need somebody to help me plug in auth calls, fix validation, ecc. But I’ll know better later.

**Comment by piotrf** [comment]
I wonder what browsers should I test this stuff on? 

Btw, here’s a sneak peek:
![kapture 2018-05-15 at 12 12 22](https://user-images.githubusercontent.com/377206/40050992-74a23268-5839-11e8-9fcd-257872d3f4cf.gif)

**Comment by piotrf** [comment]
To do next:

[ ] replicate all forms — I suspect this to be time consuming, and a pain but we’ll see
[ ] integrate `details` modal
[ ] navigation between signup/login/recovery flows

WIP:

![kapture 2018-05-15 at 15 43 32](https://user-images.githubusercontent.com/377206/40060428-de223360-5856-11e8-8b95-f19e1862004f.gif)

**Comment by AlexJupiter** [comment]
@piotrf looking fantastic!!

In terms of browsers, I would suggest:
- Latest desktop: Safari, Chrome, Firefox and Edge.
- Latest mobile: Chrome, Samsung, Stock Android, Safari

Ping me in Slack if you want to use my Browserstack account.

**Comment by piotrf** [comment]
I wonder who’s best to ask for help with routing between pages. I want both `Log In` and `Sign up` from the above GIF to redirect to their respective urls. In react-router it would be as simple as `<Link to={`/login`}>Login</Link>`. What’s current way of doing this in Sovereign?

**Comment by virgile-dev** [comment]
Hey @piotrf nice to see you back here 👋

**Comment by piotrf** [comment]
WIP some more, I’m importing already existing auth forms, and showing them with state change. This means that, for the time being, you won’t be able to link directly to say… login form, users will always need to go through the carousel screen first.

![kapture 2018-05-16 at 18 44 51](https://user-images.githubusercontent.com/377206/40145222-5361b848-5961-11e8-98c1-4661153b6263.gif)

**Comment by piotrf** [comment]
I think I’m done here. I’d appreciate if *y’all could now test it thoroughly* on a variety of browsers and devices, especially the login/signup/recovery flows. It’s all on `origin/feature/238-onboarding` branch. Make sure to do `npm install` first as I added a few dependencies there, mainly:

- https://github.com/akiran/react-slick for the slideshow bits
- https://github.com/reactjs/react-modal for the modal bits
- https://github.com/styled-components/styled-components for the styles

Here’s how I see it:

![kapture 2018-05-18 at 14 34 57](https://user-images.githubusercontent.com/377206/40234884-b449b208-5aa8-11e8-8e84-28800c7b7968.gif)

**Comment by AlexJupiter** [comment]
@piotrf I just tested on Chrome and it works great!! Awesome, awesome stuff. 

I realise that we designed this with only mobile in mind, do you think we need an improved desktop experience here?

@santisiri @virgile-dev what do you two think of this?

**Comment by virgile-dev** [comment]
This looks really great @piotrf congrats that's an awesome first contribution. 
I'm not able to test it locally somehow, would love to check how it renders in desktop. 
I'm checking with @santisiri how this can be merged.

**Comment by santisiri** [comment]
hi @piotrf let me know when you can do a PR for this.

**Comment by piotrf** [comment]
@santisiri This has been ready for testing for a while. I had reached out to the dev and design Slack channels to help out scouting bugs. Now I don’t know how far behind this branch is and, frankly I won’t have time to look at it in the near future.
